sockets/utils/getSocketUrlParts.js

```diff
@@ -18,36 +18,56 @@ const parseQuery = require('./parseQuery');
  * @see https://webpack.js.org/api/module-variables/#__resourcequery-webpack-specific
  */
 function getSocketUrlParts(resourceQuery) {
-  const scriptSource = getCurrentScriptSource();
-  const urlParts = url.parse(scriptSource);
-
+  let urlParts;
   /** @type {string | undefined} */
   let auth;
-  let hostname = urlParts.hostname;
-  let protocol = urlParts.protocol;
+  let hostname;
+  let protocol;
   let pathname = '/sockjs-node'; // This is hard-coded in WDS
-  let port = urlParts.port;
+  let port;
+  let authUrlParts;
+
+  if (typeof resourceQuery === 'string' && resourceQuery !== '') {
+    // taken from https://github.com/webpack/webpack-dev-server/blob/4ab1f21bc85cc1695255c739160ad00dc14375f1/client-src/default/utils/createSocketUrl.js#L11-L21
 
-  // FIXME:
-  // This is a hack to work-around `native-url`'s parse method,
-  // which filters out falsy values when concatenating the `auth` string.
-  // In reality, we need to check for both values to correctly inject them.
-  // Ref: GoogleChromeLabs/native-url#32
-  // The placeholder `baseURL` is to allow parsing of relative paths,
-  // and will have no effect if `scriptSource` is a proper URL.
-  const authUrlParts = new URL(scriptSource, 'http://foo.bar');
-  // Parse authentication credentials in case we need them
-  if (authUrlParts.username) {
-    auth = authUrlParts.username;
+    urlParts = url.parse(
```

Comment:
See this: there already is a way to parse the `resourceQuery`

https://github.com/pmmmwh/react-refresh-webpack-plugin/blob/3885807f741b97e5b744b2e6a9687aa0292b33eb/sockets/utils/getSocketUrlParts.js#L100-L103

Let’s remove it from the bottom so we don’t do it twice?

Reply:
agreed. thanks!